src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -166,6 +171,99 @@ public function iSearchForProductsWithName(string $name): void
         $this->client->filter();
     }
 
+    /**
+     * @Then I should (also) see the product attribute :attributeName with value :expectedAttribute
+     */
+    public function iShouldSeeTheProductAttributeWithValue(string $attributeName, string $expectedAttribute): void
+    {
+        $attribute = $this->getAttributeByName($attributeName);
+
+        Assert::same(
+            $attribute['value'],
+            $expectedAttribute
+        );
+    }
+
+    /**
+     * @Then /^I should(?:| also) see the product attribute "([^"]+)" with (positive|negative) value$/
+     */
+    public function iShouldSeeTheProductAttributeWithBoolean(string $attributeName, string $expectedAttribute): void
+    {
+        $attribute = $this->getAttributeByName($attributeName);
+
+        Assert::same(
+            $attribute['value'],
+            'positive' === $expectedAttribute
+        );
+    }
+
+    /**
+     * @Then /^I should(?:| also) see the product attribute "([^"]+)" with value ([^"]+)%$/
+     */
+    public function iShouldSeeTheProductAttributeWithPercentage(string $attributeName, int $expectedAttribute): void
+    {
+        $attribute = $this->getAttributeByName($attributeName);
+
+        Assert::same(
+            $attribute['value'],
+            $expectedAttribute / 100
+        );
+    }
+
+    /**
+     * @Then I should (also) see the product attribute :attributeName with value :expectedAttribute on the list
+     */
+    public function iShouldSeeTheProductAttributeWithValueOnTheList(string $attributeName, string $expectedAttribute): void
+    {
+        $attribute = $this->getAttributeByName($attributeName);
+
+        Assert::inArray(
+            $expectedAttribute,
+            $attribute['value']
+        );
+    }
+
+    /**
+     * @Then I should (also) see the product attribute :attributeName with date :expectedAttribute
+     */
+    public function iShouldSeeTheProductAttributeWithDate(string $attributeName, string $expectedAttribute): void
+    {
+        $attribute = $this->getAttributeByName($attributeName);
+
+        Assert::true(new \DateTime($attribute['value']) == new \DateTime($expectedAttribute));
+    }
+
+    /**
+     * @Then I should see :count attributes
+     */
+    public function iShouldSeeAttributes($count): void
+    {
+        Assert::same(
+            count($this->sharedStorage->get('product_attributes')),
+            (int) $count
+        );
+    }
+
+    /**
+     * @Then the first attribute should be :name
+     */
+    public function theFirstAttributeShouldBe($name): void
+    {
+        $attributes = $this->sharedStorage->get('product_attributes');
+
+        Assert::same(reset($attributes)['name'], $name);
+    }
+
+    /**
+     * @Then the last attribute should be :name
+     */
+    public function theLastAttributeShouldBe($name): void
+    {
+        $attributes = $this->sharedStorage->get('product_attributes');
+
+        Assert::same(end($attributes)['name'], $name);
```

Comment:
Possible null-pointer exception when `$attributes` empty.

Btw shouldn't phpstan find this?